### ISSUE TITLE
'file' placeholder in formatter config

### ISSUE THREAD

**Issue by seriyps** [basho/lager#318]
Documentation states, that it's possible to use `file` placeholder in log formatter config:
https://github.com/basho/lager/blob/3.1.0/README.md#custom-formatting

> The placeholders pid, file, line, module, function, and node will always exist if the parse transform is used.

But looks like it's just not implemented. It always produces `Undefined` and I can't find it anywhere in `lager_transform`.

**Comment by mrallen1** [comment]
Thanks for the report. We should get that fixed ASAP.

**Comment by linearregression** [comment]
It looks like module got problem too. It shows up at the beginning then revert back to undefined.

**Comment by mrallen1** [comment]
Wonder if this is possibly related to #309

**Comment by mrallen1** [comment]
@seriyps Yes, you are right. I spent a little while looking into this.  Are you looking for `file` to use the same value as you might expect to get from the `?FILE` macro?

@Vagabond was this something that was never implemented?  I looked in 1.2 and 2.x branches and I didn't find any support for it there either.

**Comment by seriyps** [comment]
@mrallen1 to be honest, I don't remember what was my case.

**Comment by mrallen1** [comment]
Understandable. Thanks anyway.

**Comment by seriyps** [comment]
@mrallen1 but yep, I think smth like `?FILE` should be nice. Not just `?MODULE ++ ".erl"`

**Comment by mrallen1** [comment]
Agreed. Thanks for the feedback! 🍬
